### PR TITLE
F extension load dependencys

### PR DESCRIPTION
**Pull request by MEichhoff**

### CODE REVIEW

File: src/Simplic.Package.Service/ExtensionHelper.cs [hunk 1]
```diff
@@ -7,10 +11,107 @@ namespace Simplic.Package.Service
...
+        /// </summary>
+        /// <returns>The path</returns>
+        public static string GetTempPath()
+        {
+            if (!string.IsNullOrWhiteSpace(tempPath))
+                return tempPath;
+
+            tempPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+
+            if (!Directory.Exists(tempPath))
```
Comment (simplicbe) [inline src/Simplic.Package.Service/ExtensionHelper.cs]: AFAIK you don't need this line. CreateDirectory will check this. But it is ok.

File: src/Simplic.Package.Service/ExtensionService.cs [hunk 2]
```diff
@@ -27,6 +28,14 @@ public ExtensionService(ILogService logService, IUnityContainer container)
         /// <inheritdoc/>
         public async void LoadExtensions(IList<string> extensions)
         {
+            //load all assemblies in extension folder.
+
+            foreach (var dll in Directory.GetFiles($"{Environment.CurrentDirectory}\\extension")
```
Comment (simplicbe) [inline src/Simplic.Package.Service/ExtensionService.cs:33]: better use `.ToLower()` here, in case someone write `MyAssembly.DLL`

**Review by simplicbe — CHANGES_REQUESTED** [review]

File: src/Simplic.Package.Service/ExtensionHelper.cs (hunk 1 above)
Comment (MEichhoff) [reply src/Simplic.Package.Service/ExtensionHelper.cs:35]: I,ve checked the documentation, you are right

File: src/Simplic.Package.Service/ExtensionHelper.cs [hunk 3]
```diff
@@ -7,10 +11,107 @@ namespace Simplic.Package.Service
...
+                Directory.CreateDirectory(tempPath);
```
Comment (MEichhoff) [inline src/Simplic.Package.Service/ExtensionHelper.cs]:
```suggestion
            Directory.CreateDirectory(tempPath);
```

File: src/Simplic.Package.Service/ExtensionService.cs [hunk 4]
```diff
@@ -27,6 +28,14 @@ public ExtensionService(ILogService logService, IUnityContainer container)
...
+                .Where(x => x.EndsWith(".dll")))
```
Comment (MEichhoff) [inline src/Simplic.Package.Service/ExtensionService.cs]:
```suggestion
                .Where(x => x.ToLower().EndsWith(".dll")))
```

File: src/Simplic.Package.Service/ExtensionService.cs [hunk 5]
```diff
@@ -58,13 +68,27 @@ public async void LoadExtensions(IList<string> extensions)
         }
 
         /// <inheritdoc/>
-        public async void LoadExtensionsFromBinaries(IDictionary<string, byte[]> asseblies)
+        public async void LoadExtensionsFromZipArchive(IList<string> extensions, ZipArchive archive)
         {
-            foreach (var entry in asseblies)
+            foreach (var dll in archive.Entries.Where(x =>
+                x.FullName.StartsWith("extension\\") &&
+                x.FullName.EndsWith(".dll")))
```
Comment (MEichhoff) [inline src/Simplic.Package.Service/ExtensionService.cs]:
```suggestion
                x.FullName.ToLower().EndsWith(".dll")))
```

**Review by simplicbe — APPROVED** [review]

**Pull request merged**